### DISCUSSION TITLE
Different column name

### DISCUSSION THREAD

**Discussion by emincer** [filamentphp/filament#6626] [Help] [answered]
I have a model called NewPage that has this method:
`    public function user()
    {
        return $this->belongsTo(User::class);
    }`

And in Filament I have a select that uses the relationship:

`        return $form
            ->schema([
                Forms\Components\Group::make()
                ->schema([
                    Forms\Components\TextInput::make('name')
                    ->maxLength(100),
                    Forms\Components\Select::make('user_id')

                    ->relationship('user', 'email')

                    ->searchable()
                    ->required()                    
                ])
            ]);`

But the column name in my database is `owner_id` and so the save fails as it is trying to use `user_id`

What am I doing wrong??

**Comment by emincer** [comment]
Follow up question. I've looked but can't find docs on the various available methods. I'd like to know more about the relationship method, like maybe there's an optional third parameter for column name? Or another chained method to set it? But I can't find a list of available methods. Like in Laravel docs there's a giant list of helper methods. Do we have something similar that I haven't found?

**Comment by emincer** [comment]
Sorry, bad copy/paste.... yes I am using `owner_id` in the make. I still get

`SQLSTATE[42S22]: Column not found: 1054 Unknown column 'user_id' in 'field list'`

on the save.

**Comment by emincer** [comment]
Well, I just changed the Model method to owner() instead of user and it's fine. Guess that works but I still would love to know if there's a way to override column name... and if I am missing some one-place collection of methods in the docs. The example code is AWESOME but the right method is sometimes hard to find.

**Reply by danharrin** [reply, marked as answer]
The second argument of `belongsTo()` in Laravel accepts a custom column name

**Reply by emincer** [reply]
D'oh! Thanks.
